### PR TITLE
Use NFI for more native calls

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

File: src/main/ruby/core/posix.rb [hunk 1]
```diff
@@ -14,9 +14,17 @@ module Truffle::POSIX
     :ushort => :uint16,
     :int => :sint32,
     :uint => :uint32,
+    :long => :sint64,
+    :ulong => :uint64,
   }
 
+  FS_ENCODING = Encoding.find('filesystem')
```
Comment (eregon) [inline src/main/ruby/core/posix.rb]: @nirvdrum The filesystem encoding is unlikely to change on the fly, right? :)
I don't want to call `Encoding.find('filesystem')` everytime and also it's quite long and verbose.
Maybe we should just have `Encoding::FILESYSTEM` and use that internally?

File: src/main/ruby/core/posix.rb [hunk 2]
```diff
@@ -27,37 +35,97 @@ def self.to_nfi_type(type)
...
+  attach_function :getgroups, [:int, [:gid_t]], :int
+
   attach_function :getrlimit, [:int, :pointer], :int
   attach_function :setrlimit, [:int, :pointer], :int
 
+  attach_function :getpriority, [:int, :id_t], :int
+  attach_function :setpriority, [:int, :id_t, :int], :int
+
+  # ENV-related
   attach_function :getenv_native, :getenv, [:string], :string
```
Comment (pitr-ch) [inline src/main/ruby/core/posix.rb]: `attach_function :getenv, [:string], :string, as: :getenv_native` would be more readable than 
`attach_function :getenv_native, :getenv, [:string], :string`

Comment (chrisseaton) [reply src/main/ruby/core/posix.rb]: This is gone anyway in a later commit.

Comment (pitr-ch) [reply src/main/ruby/core/posix.rb]: Ah this line is indeed gone, I actually meant to change the signature of `attach_function` to make these calls more readable when it's aliased. It's a nice to have change next time somebody edits it.

Comment (eregon) [reply src/main/ruby/core/posix.rb]: The signature with two names as first/second argument is how Ruby FFI does it: https://github.com/ffi/ffi/wiki/Basic-Usage
We don't have to stick to it but it seems simpler initially.
I agree `as:` is more readable, PR welcome.

I was thinking of passing a block do attach_function, but it just seems more complicated.

**Pull request merged**